packages/bot/src/struct/CommandHandler.ts

```diff
@@ -165,11 +209,25 @@ export class CommandHandler {
 		const path = join(dirname(fileURLToPath(import.meta.url)), '..', 'commands');
 		const files = readdirRecurse(path, { fileExtensions: ['js'] });
 
+		const subcommands: { rootcmd: string; command: Subcommand }[] = [];
 		for await (const file of files) {
 			const mod = (await import(pathToFileURL(file).toString())) as { default: CommandConstructor };
 			const command = container.resolve(mod.default);
 
-			this.commands.set(command.interactionOptions.name, command);
+			// check if command is a subcommand via seeing if the directory it's in is a subdirectory of the command directory
+			const directory = dirname(file).split('/').pop()!;
+			const isSubcommand = (cmd: Command | Subcommand | CommandWithSubcommands): cmd is Subcommand =>
+				!['commands', 'context-menus'].includes(directory) && !file.endsWith('index.js');
```

Comment:
We have that bool for a reason
```suggestion
				Boolean('containsSubcommands' in cmd && cmd.containsSubcommands);
```

Reply:
This checks if a file is a `Subcommand`, not if it's a `CommandWithSubcommands`. Unless you want the boolean indicator to be implemented for `Subcommand`s as well?

Reply:
Ah. No, that's my bad.